tensorflow/tools/ci_build/windows/cpu/pip/build_tf_windows_clang-cl.sh

```diff
@@ -0,0 +1,166 @@
+  || { echo "Failed to source bazel_test_lib.sh" >&2; exit 1; }
+
+# Recreate an empty bazelrc file under source root
+export TMP_BAZELRC=.tmp.bazelrc
+rm -f "${TMP_BAZELRC}"
+touch "${TMP_BAZELRC}"
+
+function cleanup {
+  # Remove all options in .tmp.bazelrc
+  echo "" > "${TMP_BAZELRC}"
+}
+trap cleanup EXIT
+
+PY_TEST_DIR="py_test_dir"
+
+SKIP_TEST=0
+RELEASE_BUILD=0
+TEST_TARGET="//${PY_TEST_DIR}/tensorflow/python/..."
+PROJECT_NAME=""
+EXTRA_BUILD_FLAGS=""
+EXTRA_TEST_FLAGS=""
+
+# --skip_test            Skip running tests
+# --enable_remote_cache  Add options to enable remote cache for build and test
+# --release_build        Build for release, compilation time will be longer to
+#                        ensure performance
+# --test_core_only       Use tensorflow/python/... as test target
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --tf_nightly) TF_NIGHTLY=1 ;;
+    --skip_test) SKIP_TEST=1 ;;
+    --enable_remote_cache) set_remote_cache_options ;;
+    --release_build) RELEASE_BUILD=1 ;;
+    --test_core_only) TEST_TARGET="//${PY_TEST_DIR}/tensorflow/python/..." ;;
+    --extra_build_flags)
+      shift
+      if [[ -z "$1" ]]; then
+        break
+      fi
+      EXTRA_BUILD_FLAGS="$1"
+      ;;
+    --project_name)
+      shift
+      if [[ -z "$1" ]]; then
+        break
+      fi
+      PROJECT_NAME="$1"
+      ;;
+    --extra_test_flags)
+      shift
+      if [[ -z "$1" ]]; then
+        break
+      fi
+      EXTRA_TEST_FLAGS="$1"
+      ;;
+    *)
+  esac
+  shift
+done
+
+if [[ "$RELEASE_BUILD" == 1 ]]; then
+  # Overriding eigen strong inline speeds up the compiling of conv_grad_ops_3d.cc and conv_ops_3d.cc
+  # by 20 minutes. See https://github.com/tensorflow/tensorflow/issues/10521
+  # Because this hurts the performance of TF, we don't override it in release build.
+  export TF_OVERRIDE_EIGEN_STRONG_INLINE=0
+else
+  export TF_OVERRIDE_EIGEN_STRONG_INLINE=1
+fi
+
+if [[ "$TF_NIGHTLY" == 1 ]]; then
+  if [[ ${PROJECT_NAME} == *"2.0_preview"* ]]; then
+    python tensorflow/tools/ci_build/update_version.py --version=2.0.0 --nightly
+  else
+    python tensorflow/tools/ci_build/update_version.py --nightly
+  fi
+  if [ -z ${PROJECT_NAME} ]; then
+    EXTRA_PIP_FLAGS="--nightly_flag"
+  else
+    EXTRA_PIP_FLAGS="--project_name ${PROJECT_NAME} --nightly_flag"
+  fi
+else
+  if [[ -v PROJECT_NAME  ]]; then
+    EXTRA_PIP_FLAGS="--project_name ${PROJECT_NAME}"
+  fi
+fi
+
+# Enable short object file path to avoid long path issue on Windows.
+echo "startup --output_user_root=${TMPDIR}" >> "${TMP_BAZELRC}"
+
+if ! grep -q "import %workspace%/${TMP_BAZELRC}" .bazelrc; then
+  echo "import %workspace%/${TMP_BAZELRC}" >> .bazelrc
+fi
+
+run_configure_for_cpu_build
+
+bazel build ${EXTRA_BUILD_FLAGS}  \
+  --experimental_cc_shared_library \
+  --build_tag_filters=-no_pip,-no_windows,-windows_excluded,-no_oss,-oss_excluded,-gpu,-tpu \
+  --copt=/std:c++latest --copt=/clang:-Weverything --linkopt=/FORCE:MULTIPLE \
```

Comment:
This file seems to be copied from https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/windows/cpu/pip/build_tf_windows.sh almost as is, with a few parameters added as this one. Is it possible to avoid this duplication (like having one shell script containing all shared parts, making these extra parameters an attribute to the script?

Reply:
We're going to toss scripts from this directory anyway once they're properly integrated into the ci/official directory, which is currently ongoing, so I don't think it's worth the time to add extra logic.

Reply:
Yes, it is possible to avoid the duplication in two ways:

1. Make changes in the .bazelrc file (https://github.com/tensorflow/tensorflow/blob/master/.bazelrc).
Add extra parameters related to clang-cl such as --copt, linkopt, platforms etc 

for this, I need to test if these changes in .bazelrc do not affect the ongoing build on the MSVC platform

2. Pass extra parameters as arguments to the script build_tf_windows.sh
These changes will be internal to Intel :
https://github.com/Intel-tensorflow/tensorflow/blob/mraunak/win_ci_scripts/tensorflow/tools/ci_build/windows/cpu/pip/ci_run_build_test_clang.sh#L144

Reply:
> We're going to toss scripts from this directory anyway once they're properly integrated into the ci/official directory, which is currently ongoing, so I don't think it's worth the time to add extra logic.

Hi @vam-google and @angerson are we okay with https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/windows/cpu/pip/build_tf_windows.sh 
Please let me know if any further change is needed from my end. Thank you for your comments

Reply:
If it's possible to add all clang-unique changes to a bazelrc configuration option (like `build:win_clang`), then please go with that.

Reply:
Hi @angerson, added clang-unique changes to the .bazelrc file, please review and let me know if any further change is needed